presto-main/src/main/java/com/facebook/presto/type/ArrayType.java

```diff
@@ -65,7 +65,26 @@ public boolean isOrderable()
     @Override
     public boolean equalTo(Block leftBlock, int leftPosition, Block rightBlock, int rightPosition)
     {
-        return compareTo(leftBlock, leftPosition, rightBlock, rightPosition) == 0;
+        Block leftArray = leftBlock.getObject(leftPosition, Block.class);
+        Block rightArray = rightBlock.getObject(rightPosition, Block.class);
+
+        if (leftArray.getPositionCount() != rightArray.getPositionCount()) {
+            return false;
+        }
+
+        int len = Math.min(leftArray.getPositionCount(), rightArray.getPositionCount());
```

Comment:
They are guaranteed to be equal as you checked 3 lines above. do `int len = leftArray......` instead.